app/controllers/thredded/autocomplete_users_controller.rb

```diff
@@ -6,17 +6,27 @@ class AutocompleteUsersController < Thredded::ApplicationController
     def index
       authorize_creating PrivateTopicForm.new(user: thredded_current_user).private_topic
       users = params.key?(:q) ? users_by_prefix : users_by_ids
+
       render json: {
-        results: users.map do |user|
-          { id:         user.id,
-            name:       user.send(Thredded.user_name_column),
-            avatar_url: Thredded.avatar_url.call(user) }
-        end
+          results: users.map do |user|
+            { id:         user.id,
+              name:       display_name(user),
+              avatar_url: Thredded.avatar_url.call(user) }
+          end
       }
     end
 
     private
 
+    # Defines how the name is displayed in the autocomplete search results
+    def display_name(user)
+      if Thredded.user_display_name_method == :to_s
+        user.send(Thredded.user_name_column)
+      else
+        "#{user.send(Thredded.user_name_column)} - #{user.send(Thredded.user_display_name_method)}"
+      end
+    end
+
```

Comment:
This isn't too heavy so I could see leaving this in the controller as it is, but extracting this out to a small service class that does this work would increase testability and would also NOT increase the surface area of the controller class.  Something like `Thredded::DisplayName`. This method could get removed and the hash above could call it with something like

```ruby
{ id:         user.id,
  name:       Thredded::DisplayName.new(user).to_s,
  avatar_url: Thredded.avatar_url.call(user) }
```

Actually, now that I'm looking at it. This could take the lead of `avatar_url` and be a configurable/overridable Proc.

```ruby
{ id:         user.id,
  name:       Thredded.display_name.call(user),
  avatar_url: Thredded.avatar_url.call(user) }
```